server/auth/auth.controller.js

```diff
@@ -0,0 +1,247 @@
+  const { token } = req.params;
+  const user = await AuthModel.findOne({
+    resetPasswordToken: token,
+    resetPasswordExpire: { $gt: Date.now() }
+  });
+  if (!user) return next(new ErrorResponse('Token Expired !!', 401));
+  await user.save();
+  return res.status(200).json({ success: true, token });
+});
+
+exports.resetPassword = asyncHandler(async (req, res, next) => {
+  const { error } = resetPasswordVal(req.body);
+  if (error) return next(new ErrorResponse(error.details[0].message, 400));
+
+  const user = await AuthModel.findOne({
+    resetPasswordToken: req.params.token,
+    resetPasswordExpire: { $gt: Date.now() }
+  });
+  user.resetPasswordToken = undefined;
+  user.resetPasswordExpire = undefined;
+  user.isAccountVerified = true;
+
+  user.password = await bcrypt.hash(req.body.password, 10);
+  await user.save();
+  return res.status(200).json({ success: true });
+});
+
+exports.createNewPassword = asyncHandler(async (req, res, next) => {
+  const { oldPassword, newPassword } = req.body;
+  const { error } = createNewPasswordVal(req.body);
+  if (error) return next(new ErrorResponse(error.details[0].message, 400));
+
+  const user = await AuthModel.findById(req.user._id).select('+password');
+
+  const isMatched = await bcrypt.compare(oldPassword, user.password);
+  if (isMatched) {
+    user.password = await bcrypt.hash(newPassword, 10);
+    await user.save();
+    return res.status(200).json({ success: true });
+  }
+  return next(new ErrorResponse('Old password is not matching', 401));
+});
+
+exports.updateDetails = asyncHandler(async (req, res, next) => {
+  let { username, email, phoneNumber } = req.body;
+  if (!username) username = req.user.username;
+  if (!email) email = req.user.email;
+  if (!phoneNumber) phoneNumber = req.user.phoneNumber;
+
+  const fieldsToUpdate = { username, email, phoneNumber };
+
+  const user = await AuthModel.findByIdAndUpdate(req.user._id, fieldsToUpdate, {
+    new: true
+  });
+
+  res.status(200).json({ success: true, user });
+});
+
+exports.deactivateAccount = asyncHandler(async (req, res, next) => {
+  const { user } = req;
+  user.isAccountActive = false;
+  await user.save();
+  return res.status(200).json({ success: true, user });
+});
+
+exports.deleteAccount = asyncHandler(async (req, res, next) => {
+  await AuthModel.findByIdAndDelete({ _id: req.user._id });
+  req.user = undefined;
+  return res.status(200).json({ success: true });
+});
+
+exports.tokenRefresher = asyncHandler(async (req, res, next) => {
+  const refreshToken = req.body.refreshToken;
+
+  try {
+    const isBlackListed = await redis.get(`BlackListed${refreshToken}`);
+    if (isBlackListed) {
+      return next(new ErrorResponse('This Token is already used', 403));
+    }
+    const decoded = await jwt.verify(refreshToken, config.jwtRefreshKey);
+    const user = await AuthModel.findById(decoded.id);
+    if (!user) return next(new ErrorResponse('User not found', 404));
```

Comment:
Do you need these lines of code? I don't think so. why do I need to know if a user exist or not during refresh my token?

Reply:
ok. Changed